test/integration/apiserver/cel/validatingadmissionpolicy_test.go

```diff
@@ -643,3 +643,84 @@ func checkFailureReason(t *testing.T, err error, expectedReason metav1.StatusRea
 		t.Error("unexpected error reason")
 	}
 }
+
+// Test_PolicyExemption tests that ValidatingAdmissionPolicy and ValidatingAdmissionPolicyBinding resources
+// are exempt from policy rules.
+func Test_PolicyExemption(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.CELValidatingAdmission, true)()
+	server, err := apiservertesting.StartTestServer(t, nil, []string{
+		"--enable-admission-plugins", "ValidatingAdmissionPolicy",
+	}, framework.SharedEtcd())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer server.TearDownFn()
+
+	config := server.ClientConfig
+
+	client, err := clientset.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	policy := makePolicy("test-policy")
+	policy.Spec.MatchConstraints = &admissionregistrationv1alpha1.MatchResources{
+		ResourceRules: []admissionregistrationv1alpha1.NamedRuleWithOperations{
+			{
+				RuleWithOperations: admissionregistrationv1alpha1.RuleWithOperations{
+					Operations: []admissionregistrationv1.OperationType{
+						"*",
+					},
+					Rule: admissionregistrationv1.Rule{
+						APIGroups: []string{
+							"*",
+						},
+						APIVersions: []string{
+							"*",
+						},
+						Resources: []string{
+							"*",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	policy.Spec.Validations = []admissionregistrationv1alpha1.Validation{{
+		Expression: "false",
+		Message:    "marker denied; policy is ready",
+	}}
+
+	policy, err = client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicies().Create(context.TODO(), policy, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	policyBinding := makeBinding("test-policy-binding", "test-policy", "")
+	if err := createAndWaitReady(t, client, policyBinding, nil); err != nil {
+		t.Fatal(err)
+	}
+
+	// validate that operations to ValidatingAdmissionPolicy are exempt from an existing policy that catches all resources
+	policyCopy := policy.DeepCopy()
+	ignoreFailurePolicy := admissionregistrationv1alpha1.Ignore
+	policyCopy.Spec.FailurePolicy = &ignoreFailurePolicy
+	_, err = client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicies().Update(context.TODO(), policyCopy, metav1.UpdateOptions{})
+	if err != nil {
+		t.Error(err)
+	}
+
+	policyBinding, err = client.AdmissionregistrationV1alpha1().ValidatingAdmissionPolicyBindings().Get(context.TODO(), policyBinding.Name, metav1.GetOptions{})
```

Comment:
I did a quick sanity check on this by updating isPolicyResource to use dummy resource names and checking that this test would fail otherwise:
```
    validatingadmissionpolicy_test.go:487: error deleting marker: endpoints "test-marker" is forbidden: ValidatingAdmissionPolicy 'test-policy' with binding 'test-policy-binding' denied request: marker denied; policy is ready
    validatingadmissionpolicy_test.go:711: validatingadmissionpolicies.admissionregistration.k8s.io "test-policy" is forbidden: ValidatingAdmissionPolicy 'test-policy' with binding 'test-policy-binding' denied request: marker denied; policy is ready
    validatingadmissionpolicy_test.go:724: validatingadmissionpolicybindings.admissionregistration.k8s.io "test-policy-binding" is forbidden: ValidatingAdmissionPolicy 'test-policy' with binding 'test-policy-binding' denied request: marker denied; policy is ready
```

Reply:
(in addition to `createAndWaitReady` which checks the marker resource)